src/Units/UnitContext.java

```diff
@@ -208,6 +237,25 @@ public int calculateCapturePower()
     return captureProgress;
   }
 
+  public int calculateVision()
+  {
+    visionRange = model.visionRange;
+    visionPierces = model.visionPierces;
+    if( null != env )
+    {
+      // if it's a surface unit, give it the boost the terrain would provide, so long as it's not adjacent-only vision
+      if( model.isSurfaceUnit() )
+        visionRange += env.terrainType.getVisionBoost();
+      if( env.weatherType == Weathers.RAIN )
+        --visionRange;
```

Comment:
This code doesn't appear to have an accommodation for "adjacent-only" vision, and I'm not sure when that's even relevant.

Reply:
That's in the next PR. DoR rain doesn't penalize movement, so it's a different weather type.

FWIW, I decided that DS and DoR rain are the same weather just acting differently in each fog mode (since both cause fog and penalize vision with no other effects).

Reply:
Okay, maybe I'll go complain about it in that PR then.